### PR TITLE
[NVIDIA XLA GPU] add helper to hlo computation to identify to_apply regions

### PR DESCRIPTION
**Pull request by Tixxx**
add helper to hlo computation to identify to_apply regions of collectives

### CODE REVIEW

File: xla/hlo/ir/hlo_computation.h [hunk 1]
```diff
@@ -813,6 +833,13 @@ class HloComputation {
   // Determines whether this computation is a custom-call computation.
   bool is_custom_call_computation_;
 
+  // If this computation is a collective sub-computation, this field points to
+  // the corresponding to_apply instruction. Otherwise, this is null.
```
Comment (jurahul) [inline xla/hlo/ir/hlo_computation.h:837]: nit: corresponding collective instruction.

File: xla/hlo/ir/hlo_instructions.cc [hunk 2]
```diff
@@ -917,6 +917,7 @@ HloAllReduceInstructionBase::HloAllReduceInstructionBase(
                                constrain_layout, channel_id),
       use_global_device_ids_(use_global_device_ids) {
   AppendComputation(reduce_computation);
+  reduce_computation->SetCollectiveCallInstruction(this);
```
Comment (jurahul) [inline xla/hlo/ir/hlo_instructions.cc:920]: I am assuming this covers both AR and RS. Can we add a test as well? in hlo_instruction_test.cc

**Review by jurahul — CHANGES_REQUESTED** [review]
Overall LGTM, just 2 minor asks.

File: xla/hlo/ir/hlo_instructions.cc (hunk 2 above)
Comment (Tixxx) [reply xla/hlo/ir/hlo_instructions.cc:920]: Yes, added tests for each op

**Review by jurahul — APPROVED** [review]

**Pull request closed without merging**